trillian/ctfe/ct_server/main.go

```diff
@@ -223,8 +218,18 @@ func main() {
 		}
 	}
 
-	err = http.ListenAndServe(*httpEndpoint, handler)
-	glog.Warningf("Server exited: %v", err)
+	// Bring up the HTTP server and serve until we get a signal not to.
+	srv := http.Server{Addr: *httpEndpoint, Handler: handler}
+	go awaitSignal(func() {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+		defer cancel()
+		srv.Shutdown(ctx)
```

Comment:
Aside: this change loses the process return code being non-zero... but that's probably a good thing, as the previous `os.Exit(1)` would have skipped any deferred functions, such as the etcd-unannouncement.

Is there an easy way of exiting a Go process with non-zero rc, but still running deferred functions?

Reply:
Good point about the defer. Maybe we can argue that responding to an instruction to quit isn't an error?

Reply:
SGTM